### PR TITLE
App (chart) customization support

### PR DESCRIPTION
**Pull request by andreas-kupries**
Ref: https://github.com/epinio/epinio/issues/1252

feat: extended app CRD to hold customization settings
feat: extended chart CRD to declare user-visible settings to customize deployment

### CODE REVIEW

File: api/v1/appchart_types.go [hunk 1]
```diff
@@ -41,6 +41,28 @@ type AppChartSpec struct {
...
+	// Settings declares the fields underneath `userValues` the user is
+	// allowed to customize when deploying an application with the helm
+	// chart referenced by this app chart.
+	Settings map[string]AppChartSetting `json:"settings,omitempty"`
+
+	// To expand and clarify the above a bit more:
+	//
+	// - Values is the app chart configuring the helm chart. This enables the
+	//   creation of multiple app charts based on the same helm chart, as
+	//   predefined setups the user can select from.
```
Comment (jimmykarily) [inline api/v1/appchart_types.go]: Maybe we should mention that this field is meant to be used by operators in order to change the default (upstream) behavior of the chart.

File: api/v1/appchart_types.go [hunk 2]
```diff
@@ -41,6 +41,28 @@ type AppChartSpec struct {
...
+	//
+	// - Settings is telling the client(s) which config fields of the helm
+	//   chart the user is allowed to change. I.e. the knobs the user has to
+	//   tune the deployment with the given app chart.
```
Comment (jimmykarily) [inline api/v1/appchart_types.go]: And here, instead of `user` maybe we should use the term `developer` or at least put it in parenthesis.

**Review by jimmykarily — APPROVED** [review]

File: api/v1/appchart_types.go (hunk 2 above)
Comment (andreas-kupries) [reply api/v1/appchart_types.go]: - `application developer`, to make it completely clear

File: api/v1/appchart_types.go (hunk 1 above)
Comment (andreas-kupries) [reply api/v1/appchart_types.go]: agree

Comment (andreas-kupries) [reply api/v1/appchart_types.go]: Done dd4edb2dacd708c1efc6cc517cb5f67cffba4c76

File: api/v1/appchart_types.go (hunk 2 above)
Comment (andreas-kupries) [reply api/v1/appchart_types.go]: Done dd4edb2dacd708c1efc6cc517cb5f67cffba4c76

**Pull request merged**